### ISSUE TITLE
fix/update_cnn - Debug cnn.py

### ISSUE THREAD

**Issue by ggutierrez545** [ggutierrez545/glearn#5]
### Bug

When training a ConvolutionalNeuralNet, issues arise with loss calculations in final layer of the network. Appears to be connected with the softmax / cross-entropy loss implementation. 

C'mon @ggutierrez545 get it together

**Comment by ggutierrez545** [comment]
Update

Appears that issue might actually lie with the way partial derivatives are calculated for ConvolutionLayers in backpropagation. Need to pay closer attention to those calculations to see if this hunch is correct.

**Comment by ggutierrez545** [comment]
WE FINALLY DID IT.

Good job me, very proud.
